### PR TITLE
dependency upgrades and a few reducer tweaks

### PR DESCRIPTION
**Pull request by aliceriot**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?

no ticket

#### What's this PR do?

I ran `yarn upgrade` and added the `@reduxjs/toolkit` package to simplify a little bit of our reducer code. I added it over in bootcamps (https://github.com/mitodl/bootcamp-ecommerce/pull/503) and thought we might want to have it over here too. It makes it really easy to create reducers! for instance, here's what I did in bootcamps to create a reducer for the drawer I added:

```js
export const toggleDrawer = createAction("TOGGLE_DRAWER")

export const drawer = createReducer(
  { showDrawer: false },
  {
    [toggleDrawer]: state => {
      state.showDrawer = !state.showDrawer
    }
  }
)
```

basically you pass to `createReducer` some default state and then a map of actions to reducer functions which are each passed a `state` object. this state object is an [immer object](https://github.com/immerjs/immer), so you can just modify it directly however you want, which is nice.

anyhow, I didn't go and rewrite all of our code or anything like that but just updated a few things to see how we can use it going forward.

#### How should this be manually tested?

just confirm that nothing broke :)

### CODE REVIEW

**Comment by abeglova** [comment]
On a mac I get
```
watch_1    | info fsevents@1.2.13: The platform "linux" is incompatible with this module.
watch_1    | info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
watch_1    | [3/4] Linking dependencies...
watch_1    | warning " > @mitodl/mdl-react-components@0.0.3" has incorrect peer dependency "@material/button@^0.26.0".
watch_1    | warning " > @mitodl/mdl-react-components@0.0.3" has incorrect peer dependency "@material/dialog@^0.26.0".
watch_1    | warning " > @mitodl/mdl-react-components@0.0.3" has incorrect peer dependency "react@^15.6.0".
watch_1    | warning " > @mitodl/mdl-react-components@0.0.3" has incorrect peer dependency "react-dom@^15.6.0".
watch_1    | warning " > redux-query-react@3.3.1" has incorrect peer dependency "react-redux@7.1.0".
watch_1    | warning " > webpack-dev-middleware@1.12.2" has incorrect peer dependency "webpack@^1.0.0 || ^2.0.0 || ^3.0.0".
```
And the site does not load

**Comment by aliceriot** [comment]
that means the yarn install process is still in progress - does it work if you give it a minute or two to spin up?

**Comment by abeglova** [comment]
Works great!
